airflow/dags/libs/news_article/db_update/article_final_selection.py

```
@@ -9,56 +9,160 @@
+        """
+        cursor.execute(query, (article_id,))
+        category_names = [row["category_name"] for row in cursor.fetchall()]
+
+        # 2. JSON 문자열 변환
+        categories = json.dumps(category_names, ensure_ascii=False)
+        
+        # 필수 필드 검증
+        required_fields = ["url", "title","sub_col", "content_col", "author", "publish_time"]
+        for field in required_fields:
+            if not article_content.get(field):
+                raise ValueError(f"[article_id={article_id}] 필수 필드 누락: {field}")
```

Ruff (0.12.2)

Abstract `raise` to an inner function

(TRY301)